library/rendering/entity_models/src/main/java/org/quiltmc/qsl/rendering/entity_models/api/Codecs.java

```diff
@@ -0,0 +1,167 @@
+    public static final class Animations {
+        public static final Codec<AnimationKeyframe> KEYFRAME = RecordCodecBuilder.create(instance -> instance.group(
+                Codec.floatRange(0, Float.MAX_VALUE).fieldOf("timestamp").forGetter(AnimationKeyframe::timestamp),
+                Vec3f.CODEC.fieldOf("transformation").forGetter(AnimationKeyframe::transformation),
+                Codec.STRING.flatXmap(
+                        s -> AnimationUtils.getInterpolatorFromName(s).map(DataResult::success).orElseGet(() -> DataResult.error("Unknown interpolator: " + s)),
+                        i -> AnimationUtils.getNameForInterpolator(i).map(DataResult::success).orElse(DataResult.error("Unknown interpolator"))
+                ).fieldOf("interpolator").forGetter(AnimationKeyframe::interpolator)
+        ).apply(instance, AnimationKeyframe::new));
+
+        public static final Codec<PartAnimation> PART_ANIMATION = RecordCodecBuilder.create(instance -> instance.group(
+                Codec.STRING.flatXmap(s -> switch (s) {
+                    case "TRANSLATE" -> DataResult.success(PartAnimation.AnimationTargets.TRANSLATE);
+                    case "ROTATE" -> DataResult.success(PartAnimation.AnimationTargets.ROTATE);
+                    case "SCALE" -> DataResult.success(PartAnimation.AnimationTargets.SCALE);
+                    default -> DataResult.error("Unknown transformation: " + s);
+                }, transformation -> {
+                    if (transformation == PartAnimation.AnimationTargets.TRANSLATE) {
+                        return DataResult.success("TRANSLATE");
+                    } else if (transformation == PartAnimation.AnimationTargets.ROTATE) {
+                        return DataResult.success("ROTATE");
+                    } else if (transformation == PartAnimation.AnimationTargets.SCALE) {
+                        return DataResult.success("SCALE");
+                    } else {
+                        return DataResult.error("Unknown transformation");
+                    }
+                }).fieldOf("transformation").forGetter(PartAnimation::transformation),
+                Codec.list(KEYFRAME).xmap(list -> list.toArray(AnimationKeyframe[]::new), Arrays::asList).fieldOf("keyframes").forGetter(PartAnimation::keyframes)
+        ).apply(instance, PartAnimation::new));
+
+        public static final Codec<Animation> ANIMATION = RecordCodecBuilder.create(instance -> instance.group(
+                Codec.floatRange(0, Float.MAX_VALUE).fieldOf("length").forGetter(Animation::length),
+                Codec.BOOL.fieldOf("looping").forGetter(Animation::looping),
+                Codec.unboundedMap(Codec.STRING, Codec.list(PART_ANIMATION)).fieldOf("animations").forGetter(Animation::animations)
+        ).apply(instance, Animation::new));
+    }
+
+    public static final class Model {
+        public static final Codec<TextureDimensions> TEXTURE_DIMENSIONS = RecordCodecBuilder.create((instance) ->
```

Comment:
the lambdas for the `RecordCodecBuilder.create(...)` are inconsistent. Some have the parentheses around the single argument while others don't. Also, some start `instance.group(...)` on the same line and others on a new line.

Closing parentheses for the `RecordCodecBuilder.create(...)` are also inconsistent, some are same line, others are new line